src/app/(pages)/mobility.tsx

```diff
@@ -26,9 +29,29 @@ export default function Mobility(): JSX.Element {
     }
     const [error, setError] = useState<Error | null>(null)
     const [loadingState, setLoadingState] = useState(LoadingState.LOADING)
-    const { mobilityKind, mobilityStation } = React.useContext(MobilityContext)
+    const { mobilityKind, toggleMobilityKind } =
+        React.useContext(MobilityContext)
+    const { mobilityStation, toggleMobilityStation } =
+        React.useContext(MobilityContext)
+
+    const displayTypes = ['Bus', 'Train', 'Car', 'EV']
+    const busLocations = [{}]
+    const trainLocations = [{}]
+    trainLocations.pop()
+    busLocations.pop()
+    MobilityJSON.bus.stations.map((station) =>
+        busLocations.push({ key: station.id, value: station.name })
+    )
+    MobilityJSON.train.stations.map((station) =>
+        trainLocations.push({ key: station.id, value: station.name })
+    )
```

Comment:
You can shorten the code significantly and make it more efficient if you use the `map`` function instead.

```js
const busLocations = MobilityJSON.bus.stations.map((station) => ({
    key: station.id,
    value: station.name,
}))
```